### PR TITLE
chore: update requirement file in metadata

### PR DESCRIPTION
**Pull request by feng-tao**
Signed-off-by: feng-tao <fengtao04@gmail.com>

### Summary of Changes

Put additional proxy requirement into setup.py extra section.

Some contexts: https://amundsenworkspace.slack.com/archives/CHQERT0D7/p1622780101143900, no RC of the issue yet

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by feng-tao** [comment]
cc @mgorsk1 @crazy-2020

**Review by crazy-2020 — APPROVED** [review]

**Comment by mgorsk1** [comment]
this makes it more difficult to use default docker images with atlas and rds (for neo4j you don't need to do anything else, for atlas and rds you will need to prebuild i'd additionally). maybe we should think about publishing separate docker images for each backend so the users are not affected by this. wdyt?

File: metadata/setup.py [hunk 1]
```diff
@@ -25,7 +25,12 @@
     install_requires=requirements,
     extras_require={
         'oidc': ['flaskoidc==0.2.3'],
-        'atlas': ['apache-atlas==0.0.11']
+        'atlas': ['apache-atlas==0.0.11',
+                  'amundsenatlastypes==1.2.0'],
```
Comment (mgorsk1) [inline metadata/setup.py:29]: most recent is 1.2.2

**Comment by feng-tao** [comment]
> this makes it more difficult to use official docker images with atlas and rds (for neo4j you don't need to do anything else, for atlas and rds you will need to prebuild additionally). Dockerfile should still build with all deps. long term maybe we should think about publishing separate docker images for each backend so the users are not affected by this. wdyt?

+1 on separate docker image for different proxy

File: metadata/setup.py (hunk 1 above)
Comment (feng-tao) [reply metadata/setup.py]: update

**Comment by mgorsk1** [comment]
only thing then is to figure out if we have N * 2 images (N - number of proxies times two cause there is one with and one wihtout oidc right now)

**Review by allisonsuarez — APPROVED** [review]
LGTM

**Pull request merged**